src/main/java/codeu/model/store/basic/MessageStore.java

```diff
@@ -106,11 +130,10 @@ public void addMessage(Message message) {
   /** Access the current set of Messages sent by a specific user. */
   public List<Message> getMessagesByAuthor(UUID authorId) {
 
-    List<Message> authorMessages = new ArrayList<>(messages);
-
-    //Removes messages from the list if their authorId doesn't match to the one given
-    authorMessages.removeIf(message-> !message.getAuthorId().equals(authorId));
-
+    if (!messagesByAuthorId.containsKey(authorId)) {
+      return null;
```

Comment:
Better to return an empty list, this could cause a NPE somewhere.

In cases where you do want to return null, you should annotate the method with `@Nullable`